orangewidget/gui.py

```diff
@@ -1669,6 +1670,26 @@ def comboBox(widget, master, value, box=None, label=None, labelWidth=None,
     return combo
 
 
+def deferred(func):
+    @wraps(func)
+    def _commit_replacement():
+        raise RuntimeError(
+            "This function is deferred; explicitly call "
+            f"{func.__name__}.deferred or {func.__name__}.now")
```

Comment:
If I understand this correctly `self.commit.deferred` is equivalent to the old`self.commit` and `self.commit.now` to `self.unconditional_commit`. In that case, I would remove the `or {func.__name__}.now` from the exception message and create a separate RuntimeError for the unconditional_commit.

Reply:
Your point is, to my understanding, that previously such call resulted in the deferred call. Skipping `or {func.__name__}.now` would be the correct instruction for converting the old code, but misleading for new code(rs), where both calls are potentially correct.

I guess I'd keep the message as it is.